controllers/managedcluster_controller.go

```diff
@@ -0,0 +1,146 @@
+		For(&clusterv1.ManagedCluster{}, builder.WithPredicates(predicate.GenerationChangedPredicate{})).
+		Watches(
+			&viewv1beta1.ManagedClusterView{},
+			handler.EnqueueRequestsFromMapFunc(r.managedClusterViewRequestMapper),
+			builder.WithPredicates(managedClusterViewStatusChangePredicate, predicate.GenerationChangedPredicate{}),
+		).
+		Complete(r)
+}
+
+func (r *ManagedClusterReconciler) managedClusterViewRequestMapper(ctx context.Context, obj client.Object) []reconcile.Request {
+	mcv, ok := obj.(*viewv1beta1.ManagedClusterView)
+
+	if !ok {
+		r.Logger.Error("Received object is not a ManagedClusterView object")
+		return []reconcile.Request{}
+	}
+	logger := r.Logger.With("ManagedClusterView", mcv.Name)
+	mcName, ok := mcv.Labels[utils.MCVLabelKey]
+
+	if !ok {
+		return []reconcile.Request{}
+	}
+
+	logger.Debug("ManagedClusterView of interest just got updated. Raising a reconcile request for ManagedCluster", "ManagedCluster", mcName)
+
+	return []reconcile.Request{
+		{
+			NamespacedName: types.NamespacedName{
+				Name: mcName,
+			},
+		},
+	}
+
+}
+func (r *ManagedClusterReconciler) processManagedClusterViews(ctx context.Context, managedCluster clusterv1.ManagedCluster) error {
+	resourceType := "ConfigMap"
+	odfInfoConfigMapNamespacedName, err := getNamespacedNameForClusterInfo(managedCluster)
+	if err != nil {
+		r.Logger.Error("Error while getting the namespaced name of the configmap")
+		return err
+	}
+	mcv, or, err := utils.CreateOrUpdateManagedClusterView(ctx, r.Client, odfInfoConfigMapNamespacedName.Name, odfInfoConfigMapNamespacedName.Namespace, resourceType, managedCluster.Name)
+	if err != nil {
+		r.Logger.Error("Failed to create or update ManagedClusterView", "error", err)
+		return err
+	}
+	r.Logger.Info(fmt.Sprintf("ManagedClusterView was %s", or), "ManagedClusterView", mcv.Name)
+
+	return nil
+}
+
+func getNamespacedNameForClusterInfo(managedCluster clusterv1.ManagedCluster) (types.NamespacedName, error) {
+	clusterClaims := managedCluster.Status.ClusterClaims
+	for _, claim := range clusterClaims {
+		if claim.Name == OdfInfoNamespacedNameClaimName {
+			namespacedName := strings.Split(claim.Value, "/")
+			if len(namespacedName) != 2 {
+				return types.NamespacedName{}, fmt.Errorf("invalid format for namespaced name claim: expected 'namespace/name', got '%s'", claim.Value)
+			}
+			return types.NamespacedName{Namespace: namespacedName[0], Name: namespacedName[1]}, nil
+		}
+	}
+
+	return types.NamespacedName{}, errors.New("cannot find namespaced name claim in cluster status")
+}
+
+var managedClusterViewStatusChangePredicate = predicate.Funcs{
+	UpdateFunc: func(e event.UpdateEvent) bool {
+		oldObj, okOld := e.ObjectOld.(*viewv1beta1.ManagedClusterView)
+		newObj, okNew := e.ObjectNew.(*viewv1beta1.ManagedClusterView)
+		if !okOld || !okNew {
+			return false
+		}
+
+		return !reflect.DeepEqual(oldObj.Status, newObj.Status)
+	},
+	CreateFunc: func(e event.CreateEvent) bool {
+		return true
+	},
```

Comment:
Since we're the ones creating it, we don't need to reconcile create. We only need to reconcile when someone else changes it.

Reply:
done